File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/CopySample/src/com/hp/workpath/sample/copysample/task/DeleteStoredTask.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.copysample.task;

import android.os.Handler;
import android.os.Looper;

import com.hp.workpath.api.Result;
import com.hp.workpath.api.copier.CopierService;
import com.hp.workpath.api.copier.JobCredentialsAttributes;
import com.hp.workpath.sample.copysample.Logger;
import com.hp.workpath.sample.copysample.MainActivity;
import com.hp.workpath.sample.copysample.R;

import java.lang.ref.WeakReference;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class DeleteStoredTask {

    private final WeakReference<MainActivity> mContextRef;
    private JobCredentialsAttributes jobCredentials;

    private String mErrorMsg;
    private Result result;

    private ExecutorService executor = Executors.newSingleThreadExecutor();
    private Handler handler = new Handler(Looper.getMainLooper());

    public DeleteStoredTask(final MainActivity context, JobCredentialsAttributes jobCredentials) {
        this.mContextRef = new WeakReference<>(context);
        this.jobCredentials = jobCredentials;
        this.result = new Result();
    }

    public void taskExecute(final String... params) {
        try {
            executor.execute(new Runnable() {
                @Override
                public void run() {
                    MainActivity activity = mContextRef.get();
                    try {
                        String storedJobId = params[0];
                        CopierService.deleteStoredJob(activity, storedJobId, jobCredentials, result);
                    } catch (IllegalArgumentException iae) {
                        mErrorMsg = "IllegalArgumentException";
                        Logger.showResult(null, mErrorMsg + " " + iae.getMessage());
                        executor.shutdown();
                    } catch (Throwable t) {
                        mErrorMsg = "Unknown exception";
                        Logger.showResult(null, mErrorMsg + " " + t.getMessage());
                        executor.shutdown();
                    }
                    onPostExecute();
                }
            });
        } catch (Exception e) {
            mErrorMsg = "Unknown exception";
            Logger.showResult(null, mErrorMsg + " " + e.getMessage());
            executor.shutdown();
        }
    }

    private void onPostExecute() {
        handler.post(new Runnable() {
            @Override
            public void run() {
                if (result.getCode() != Result.RESULT_OK) {
                    Logger.showResult(mContextRef.get(), "CopierService.deleteStoredJob", result);
                } else {
                    Logger.showResult(mContextRef.get(), mContextRef.get().getString(R.string.job_deleted));
                }
            }
        });
    }
} 

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/CopySample/src/com/hp/workpath/sample/copysample/task/DeleteStoredTask.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.copysample.task

import com.hp.workpath.api.Result
import com.hp.workpath.api.copier.CopierService
import com.hp.workpath.api.copier.JobCredentialsAttributes
import com.hp.workpath.sample.copysample.Logger
import com.hp.workpath.sample.copysample.MainActivity
import com.hp.workpath.sample.copysample.R
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.lang.ref.WeakReference

class DeleteStoredTask(context: MainActivity, jobCredentials: JobCredentialsAttributes) {
    private val mContextRef: WeakReference<MainActivity> = WeakReference(context)
    private val jobCredentials: JobCredentialsAttributes = jobCredentials
    private lateinit var mErrorMsg: String
    private var mThrowable: Throwable? = null
    private val result: Result = Result()

    suspend fun execute(vararg params: String) {
        try {
            val storedJobId = params[0]
            mContextRef.get()?.run {
                CopierService.deleteStoredJob(this, storedJobId, jobCredentials, result)
            }
        } catch (iae: IllegalArgumentException) {
            mErrorMsg = "IllegalArgumentException"
            mThrowable = iae
        } catch (t: Throwable) {
            mErrorMsg = "Unknown exception";
            mThrowable = t;
        }
        onPostExecute()
    }

    private suspend fun onPostExecute() {
        withContext(Dispatchers.Main) {
            mContextRef.get()?.run {
                if (::mErrorMsg.isInitialized) {
                    Logger.showResult(this, "$mErrorMsg, ${mThrowable?.message}")
                } else if (result.code != Result.RESULT_OK) {
                    Logger.showResult(this, "CopierService.deleteStoredJob", result)
                } else {
                    Logger.showResult(this, this.getString(R.string.job_deleted))
                }
            }
        }
    }
}